app/server/runtime/src/main/java/io/syndesis/server/runtime/DefaultMigrator.java

```diff
@@ -15,56 +15,74 @@
  */
 package io.syndesis.server.runtime;
 
-import java.io.IOException;
+import static io.syndesis.common.util.Json.map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.stereotype.Service;
 
 import javax.script.ScriptEngine;
 import javax.script.ScriptEngineManager;
 import javax.script.ScriptException;
 
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Paths;
+
 import io.syndesis.common.util.IOStreams;
 import io.syndesis.common.util.Resources;
 import io.syndesis.common.util.SyndesisServerException;
-import io.syndesis.server.jsondb.WithGlobalTransaction;
 import io.syndesis.server.jsondb.JsonDB;
+import io.syndesis.server.jsondb.WithGlobalTransaction;
 import io.syndesis.server.jsondb.dao.Migrator;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.core.io.Resource;
-import org.springframework.core.io.ResourceLoader;
-import org.springframework.stereotype.Service;
-
-import static io.syndesis.common.util.Json.map;
-
 @Service
 public class DefaultMigrator implements Migrator {
     private static final Logger LOG = LoggerFactory.getLogger(DefaultMigrator.class);
-
+    private String migrationDir;
     private final ResourceLoader resourceLoader;
 
     public DefaultMigrator(final ResourceLoader resourceLoader) {
         this.resourceLoader = resourceLoader;
     }
 
     @Override
-    public void migrate(final JsonDB jsondb, final int toVersion) {
+    public void migrate(final JsonDB jsondb, final int schema) {
         if (jsondb instanceof WithGlobalTransaction) {
-            ((WithGlobalTransaction) jsondb).withGlobalTransaction(checkpointed -> performMigration(checkpointed, toVersion));
+            ((WithGlobalTransaction) jsondb).withGlobalTransaction(checkpointed -> performMigration(checkpointed, schema));
         } else {
-            performMigration(jsondb, toVersion);
+            performMigration(jsondb, schema);
         }
     }
 
+    @Override
+    public void migrate(JsonDB jsondb, int schema, String migrationDir) {
+        this.migrationDir = migrationDir;
+        migrate(jsondb, schema);
+    }
+
     private void performMigration(final JsonDB utilized, final int toVersion) {
         final String path = migrationsScriptPrefix() + toVersion + ".js";
```

Comment:
...and loop over `baseResources[n]` so we can support from multiple base resources also, first one that locates the resource wins; I think that was the gist of the change...